dashboard/app/controllers/policy_compliance_controller.rb

```diff
@@ -18,13 +20,73 @@ def child_account_consent
     user = permission_request.user
     #Update User
     if user.child_account_compliance_state != User::ChildAccountCompliance::PERMISSION_GRANTED
-      user.child_account_compliance_state = User::ChildAccountCompliance::PERMISSION_GRANTED
-      user.child_account_compliance_state_last_updated = DateTime.now.new_offset(0)
+      user.update_child_account_compliance(User::ChildAccountCompliance::PERMISSION_GRANTED)
       user.save!
       parent_email = permission_request.parent_email
       ParentMailer.parent_permission_confirmation(parent_email).deliver_now
     end
     @permission_granted = true
     @permission_granted_date = user.child_account_compliance_state_last_updated
   end
+
+  # POST /policy_compliance/child_account_consent
+  # This URL creates a parental permission request. It is invoked via the
+  # lockout panel or potentially user settings by the child account.
+  def child_account_consent_request
+    # If we already comply, don't suddenly invalid it
+    if current_user.child_account_compliance_state == User::ChildAccountCompliance::PERMISSION_GRANTED
+      redirect_back fallback_location: lockout_path and return
+    end
+
+    # Only allow three unique permission requests per day
+    date = Time.zone.today
+    permission_requests = ParentalPermissionRequest.where(
+      user: current_user,
+      created_at: date.midnight..date.end_of_day
+    ).limit(3).count
+
+    # Create a ParentalPermissionRequest token for user and parent email
+    # When the student 'updates' the parental email, we actually just create a
+    # new request row.
+    permission_request = ParentalPermissionRequest.find_or_initialize_by(
+      user: current_user,
+      parent_email: params.require(:'parent-email')
+    )
+
+    # If we are making a new request but already sent too many today,
+    # just bail and return whence we came
+    if permission_request.new_record? && permission_requests >= 3
+      redirect_back fallback_location: lockout_path and return
+    end
+
+    # If this is not a new row, we are resending the email
+    if permission_request.persisted?
+      permission_request.resends_sent += 1
+    end
+
+    # Do not send more than three emails to the same email
+    if permission_request.resends_sent >= 3
+      redirect_back fallback_location: lockout_path and return
+    end
+
+    # Save (will reassign the updated_at date)
+    permission_request.save!
+
+    # Update the User
+    current_user.update_child_account_compliance(User::ChildAccountCompliance::REQUEST_SENT)
+    current_user.save!
+
+    # Send the request email
+    ParentMailer.parent_permission_request(
+      permission_request.parent_email,
+      url_for(
```

Comment:
Does this create the fully-qualified URL without specifying a hostname? The docs seem to imply it'll only return the path.

Reply:
It does indeed create a fully-qualified URL... but I'll add the `only_path: false` argument which supposedly ensures this is true. I'll add a test to also check for the fully-qualified URL.